common/utils/include/claragenomics/utils/cudautils.hpp

```diff
@@ -43,24 +43,28 @@ namespace cudautils
 /// \param code The CUDA status code of the function being asserted
 /// \param file Filename of the calling function
 /// \param line File line number of the calling function
-/// \param abort If true, hard-exit on CUDA error
-inline void gpu_assert(cudaError_t code, const char* file, int line, bool abort = false)
+/// \param cuda_sync Synchronize CUDA device before checking error
+inline void gpu_assert(cudaError_t code, const char* file, int line, bool cuda_sync = false)
```

Comment:
Do we need/want `cuda_sync` as runtime parameter? Is there a likely use case, where we will call `gpu_assert` with `cuda_sync` set in some places, but not in others?

I guess, I would introduce macro `CGA_DEVICE_SYNCHRONIZE` to switch on the device sync. Similar to `NDEBUG`, such that there is an easy way to switch it on when building the project.

Reply:
> Is there a likely use case, where we will call gpu_assert with cuda_sync set in some places, but not in others?

The use case I had in mind for this was enabling it say for `cudapoa` but not `cudaaligner` when running racon to get to the poa parts faster, since the device synchronize slows down execution.

I can go the macro route as well. That would not require any code changes to enable either, so more user friendly too

Reply:
Ok, but then we would need to generalize the `CGA_CU_CHECK_ERROR` macro, right?

Reply:
Yes it would, and I missed that :). Changed it to macro, if it proves bothersome will change back to runtime parameter later